test/extended/operators/certs.go

```diff
@@ -179,3 +200,153 @@ var _ = g.Describe("[sig-arch][Late]", func() {
+	namespaceObj := resourceread.ReadNamespaceV1OrDie(namespaceYaml)
+
+	client := kubeClient.CoreV1().Namespaces()
+	actualNamespace, err := client.Create(ctx, namespaceObj, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return "", fmt.Errorf("error creating namespace: %v", err)
+	}
+	return actualNamespace.Name, nil
+}
+
+func createServiceAccount(ctx context.Context, kubeClient kubernetes.Interface, namespace string) error {
+	serviceAccountObj := resourceread.ReadServiceAccountV1OrDie(serviceAccountYaml)
+	serviceAccountObj.Namespace = namespace
+	client := kubeClient.CoreV1().ServiceAccounts(namespace)
+	_, err := client.Create(ctx, serviceAccountObj, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return fmt.Errorf("error creating service account: %v", err)
+	}
+	return nil
+}
+
+func createRBACBindings(ctx context.Context, kubeClient kubernetes.Interface, namespace string) error {
+	roleBindingObj := resourceread.ReadRoleBindingV1OrDie(roleBindingPrivilegedYaml)
+	roleBindingObj.Namespace = namespace
+
+	client := kubeClient.RbacV1().RoleBindings(namespace)
+	_, err := client.Create(ctx, roleBindingObj, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return fmt.Errorf("error creating hostaccess SCC CRB: %v", err)
+	}
+	return nil
+}
+
+func createPods(ctx context.Context, kubeClient kubernetes.Interface, namespace string, nodeList *corev1.NodeList, testImagePullSpec string) error {
+	client := kubeClient.CoreV1().Pods(namespace)
+
+	podTemplate := resourceread.ReadPodV1OrDie(podYaml)
+	for _, node := range nodeList.Items {
+		podObj := podTemplate.DeepCopy()
+		podObj.Namespace = namespace
+		podObj.Spec.NodeName = node.Name
+		podObj.Spec.Containers[0].Image = testImagePullSpec
+
+		actualPod, err := client.Create(ctx, podObj, metav1.CreateOptions{})
+		if err != nil && !apierrors.IsAlreadyExists(err) {
+			return fmt.Errorf("error creating pod on node %s: %v", node.Name, err)
+		}
+
+		timeLimitedCtx, cancel := context.WithTimeout(ctx, time.Minute)
+		defer cancel()
+
+		if _, watchErr := watchtools.UntilWithSync(timeLimitedCtx,
+			cache.NewListWatchFromClient(
+				kubeClient.CoreV1().RESTClient(), "pods", namespace, fields.OneTermEqualSelector("metadata.name", actualPod.Name)),
+			&corev1.Pod{},
+			nil,
+			func(event watch.Event) (bool, error) {
+				pod := event.Object.(*corev1.Pod)
+				return pod.Status.Phase == corev1.PodSucceeded, nil
+			},
+		); watchErr != nil {
+			return fmt.Errorf("pod %s in namespace %s didn't complete: %v", actualPod.Name, namespace, watchErr)
+		}
+	}
+	return nil
+}
+
+func fetchNodePKIList(ctx context.Context, kubeClient kubernetes.Interface, node *corev1.Node) (*certgraphapi.PKIList, error) {
+	pkiList := &certgraphapi.PKIList{}
+
+	allBytes, err := nodeaccess.GetNodeLogFile(ctx, kubeClient, node.Name, certInspectResultFile)
```

Comment:
If memory serves, this uses a construct that raw kube clusters disallow.  While I appreciate this works, you could more generally

1. make the `collector` container into an initcontainer
2. make a `pause` container 
3. watch for `pause` to enter running state
4. make this copy pull from the pod instead of the node
5. delete the pod after a successful copy

This would
1. ensure we work on stock kube as well in the (likely) case we make this a separate command
2. avoid writing content to the host filesystem

Reply:
We use [Node Logs API](https://kubernetes.io/blog/2023/04/21/node-log-query-alpha/), its however in alpha

Reply:
> We use [Node Logs API](https://kubernetes.io/blog/2023/04/21/node-log-query-alpha/), its however in alpha

ours is patched to expose more files.

Reply:
This is still a problem.  Let's collect from the pod

Reply:
Reworked to collect json in initContainer, mount it via emptyDir to the "pause" container and fetch it via `pod/exec`